src/service/resource.js

```diff
@@ -734,6 +731,24 @@ export class Resource {
     return distance < viewportBox.height * multiplier / scrollPenalty;
   }
 
+  /**
+   * Calculates the layout's viewport distance ratio, using an iterative
+   * calculation based on tree depth and number of layer scrolls it would take
+   * to view the element.
+   *
+   * @param {number} currentScore
+   * @param {!./layers-impl.LayoutElement} layout
+   * @param {number} depth
+   * @return {number}
+   */
+  layersDistanceRatio_(currentScore, layout, depth) {
+    const depthPenalty = 1 + (depth / 10);
+    const nonActivePenalty = layout.isActiveUnsafe() ? 1 : 2;
+    const distance = layout.getHorizontalViewportsFromParent() +
+        layout.getVerticalViewportsFromParent();
+    return currentScore + (nonActivePenalty * depthPenalty * distance);
+  }
```

Comment:
Likely. We can file a followup for it, since we can probably also reuse this in  https://github.com/ampproject/amphtml/blob/master/src/service/resources-impl.js#L1708-L1751.

https://github.com/ampproject/amphtml/issues/15058